morpheus/_lib/src/io/loaders/rest.cpp

```diff
@@ -17,21 +17,295 @@
 
 #include "morpheus/io/loaders/rest.hpp"
 
+#include <boost/algorithm/string.hpp>
+#include <boost/asio.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/version.hpp>
 #include <glog/logging.h>
 #include <nlohmann/json.hpp>
+#include <pybind11/cast.h>
+#include <pybind11/gil.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/pytypes.h>
+#include <pymrc/utilities/object_cache.hpp>
 
 #include <memory>
 #include <ostream>
 #include <stdexcept>
 
+namespace py = pybind11;
+
+namespace beast = boost::beast;
+namespace http  = beast::http;
+namespace net   = boost::asio;
+using tcp       = net::ip::tcp;
+
+#define PORT "8081"
+
 namespace morpheus {
 RESTDataLoader::RESTDataLoader(nlohmann::json config) : Loader(config) {}
 
```

Comment:
As a general comment, lets be a bit more explicit about error handling in each section of the code here, and I'll make an issue to do the same for other existing loaders. In many cases, _any_ calls to python code could fail, so lets be sure all of them are wrapped in a try/catch block and exceptions are processed as expected when `processes_failures_as_errors` is false, otherwise we throw.